From 29460313656e167b0a02a3ea1c571a9338154d7c Mon Sep 17 00:00:00 2001
From: "sujuan.chen" <sujuan.chen@mediatek.com>
Date: Thu, 18 May 2023 15:01:47 +0800
Subject: [PATCH 058/146] mtk: mt76: mt7996: reset addr_elem when delete ba

The old addr element info may be used when the signature is not equel to
0xff, and sta will find error SDP cause the SDP/SDL=0 issue.

1. without this patch will delete wrong session id when delete ba.
Due to fw change the cmd format.
https://gerrit.mediatek.inc/c/neptune/firmware/bora/wifi/custom/+/7969193

mtk: mt76: mt7996: Refactor rro del ba command format

1. remove unused struct
2. refactor upstream del ba command format

Signed-off-by: Rex Lu <rex.lu@mediatek.com>
Signed-off-by: sujuan.chen <sujuan.chen@mediatek.com>
---
 mt76.h       | 1 +
 mt7996/mcu.h | 4 +++-
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/mt76.h b/mt76.h
index 76c79346..38addcea 100644
--- a/mt76.h
+++ b/mt76.h
@@ -453,6 +453,7 @@ struct mt76_rx_tid {
 	u16 nframes;
 
 	u8 num;
+	u16 session_id;
 
 	u8 started:1, stopped:1, timer_pending:1;
 
diff --git a/mt7996/mcu.h b/mt7996/mcu.h
index d0a0c595..3802699e 100644
--- a/mt7996/mcu.h
+++ b/mt7996/mcu.h
@@ -273,7 +273,9 @@ struct mt7996_mcu_wed_rro_ba_delete_event {
 	__le16 len;
 
 	__le16 session_id;
-	u8 __rsv2[2];
+	__le16 mld_id;
+	u8 tid;
+	u8 __rsv[3];
 } __packed;
 
 enum  {
-- 
2.45.2

